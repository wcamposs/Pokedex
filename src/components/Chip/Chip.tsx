// libraries
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

// js
import colors from '../../utils/colors';

// interfaces
import { ChipProps } from '../../utils/interfaces';

function Chip(props: ChipProps) {
	const { name, isMythical = false, isLegendary = false } = props;

	const typeColors: { [key: string]: string } = {
		bug: '#5e7a4b',
		dark: '#0a0a0a',
		dragon: '#ed7700',
		electric: '#dbe617',
		fairy: '#ff66c7',
		fighting: '#ffa600',
		fire: '#ff1100',
		flying: '#ffd587',
		ghost: '#8733ff',
		grass: '#00bf30',
		ground: '#a68863',
		ice: '#4ad7ff',
		normal: '#949494',
		psychic: '#46e3d7',
		poison: '#b216f0',
		rock: '#54412a',
		steel: '#737373',
		water: '#0357ff',
		mythical: '#000000',
		legendary: '#000000',
	};

	const typeColor = typeColors[name] || colors.black;
	const specialPokemon = isMythical || isLegendary;

	return (
		<View style={[styles.container, { backgroundColor: typeColor }]}>
			<Text
				style={styles.typeName(specialPokemon, typeColors['electric'])}>
				{name.toUpperCase()}
			</Text>
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		justifyContent: 'center',
		alignItems: 'center',
		height: 20,
		maxHeight: 20,
		width: 68,
		maxWidth: 100,
		borderRadius: 30,
		marginHorizontal: 4,
		textAlign: 'center',
		// flexWrap: 'wrap', // TEST WHEN POKEMON HAVE 3 OR 4 TYPES
	},
	typeName: (specialType, color) => ({
		color: specialType ? color : colors.white,
		fontSize: 10,
		fontWeight: 'bold',
	}),
});

export default Chip;
