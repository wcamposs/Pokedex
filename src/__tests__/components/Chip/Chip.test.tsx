import React from 'react';
import renderer from 'react-test-renderer';

import Chip from '../../../components/Chip/Chip';

jest.useFakeTimers();

describe('<Chip />', () => {
	it('renders correctly as a normal pokemon chip', () => {
		const tree = renderer
			.create(<Chip name="bug" isMythical={false} isLegendary={false} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly as a mythical pokemon chip', () => {
		const tree = renderer
			.create(<Chip name="bug" isMythical={true} isLegendary={false} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly as a legendary pokemon chip', () => {
		const tree = renderer
			.create(<Chip name="bug" isMythical={false} isLegendary={true} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
