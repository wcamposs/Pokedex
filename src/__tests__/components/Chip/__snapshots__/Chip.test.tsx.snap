// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> renders correctly as a legendary pokemon chip 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "height": 20,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 20,
        "maxWidth": 100,
        "textAlign": "center",
        "width": 68,
      },
      Object {
        "backgroundColor": "#5e7a4b",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#dbe617",
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    BUG
  </Text>
</View>
`;

exports[`<Chip /> renders correctly as a mythical pokemon chip 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "height": 20,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 20,
        "maxWidth": 100,
        "textAlign": "center",
        "width": 68,
      },
      Object {
        "backgroundColor": "#5e7a4b",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#dbe617",
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    BUG
  </Text>
</View>
`;

exports[`<Chip /> renders correctly as a normal pokemon chip 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "height": 20,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 20,
        "maxWidth": 100,
        "textAlign": "center",
        "width": 68,
      },
      Object {
        "backgroundColor": "#5e7a4b",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    BUG
  </Text>
</View>
`;
